cpp/CMakeLists.txt

```diff
@@ -676,8 +678,11 @@ if(NOT USE_NVTX)
   target_compile_definitions(cudf PUBLIC NVTX_DISABLE)
 endif()
 
+# Define RMM logging level
+target_compile_definitions(cudf PUBLIC "RMM_LOGGING_LEVEL=LIBCUDF_LOGGING_LEVEL")
```

Comment:
Should logging levels of RMM and cuDF be independently controlled?

Reply:
Assuming they do - how do we handle setting `SPDLOG_ACTIVE_LEVEL`? RMM sets it based on RMM_LOGGING_LEVEL, and we need to set it based on `LIBCUDF_LOGGING_LEVEL`.

Reply:
Ah. Good point. I don't have feelings on this, we can keep it all the same.

I _think_ spdlog may have multiple ways to set the logging level, and compile-time options are only one of the methods. It would be hard to ensure that debug messages are compiled out while retaining runtime control over two different loggers. 😖

Reply:
Yes, the logging level can be set the runtime. This is what we use to set the level if `LIBCUDF_LOGGING_LEVEL` env var is set.

The CMake config var can be seen as minimal logging level that the build includes, since it literally undefines logging macros for level below its value (which is great for security/perf). AFAIK, the logging level for both libcudf will stay at INFO (or whatever is the default) even if CMake config var is TRACE/DEBUG.

I'm wary of having different SPDLOG_ACTIVE_LEVEL values in RMM and libcudf as it feels like possible ODR violation 🤷

Reply:
I don't think you would run into any issues. RMM only sets this for compiled components i.e. its tests and benchmarks. The main rmm code is header-only so there are no compiled components where `SPDLOG_ACTIVE_LEVEL` would be set to conflict with whatever libcudf sets.

Reply:
Great point, yeah that makes sense. I believe RMM sets this PUBLIC, but again it's only done in tests and benchmarks so it probably doesn't matter.